tests/testthat/test-shinytest2-init.R

```diff
@@ -8,6 +8,65 @@ testthat::test_that("e2e: teal app initializes with no errors", {
   app$stop()
 })
 
+testthat::test_that("e2e: teal app initializes with sessionInfo modal", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check if button exists.
+  button_selector <- "#teal-sessionInfo-button"
+  testthat::expect_equal(
+    app$get_text(button_selector),
+    "Session Info"
+  )
+
+  app$click(selector = button_selector)
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check header and title content.
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-header > h4"),
+    "SessionInfo"
+  )
+  testthat::expect_equal(
+    app$get_text("#teal-sessionInfo-copy_button1"),
+    "Copy to Clipboard"
+  )
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
+    "Dismiss"
+  )
+
+  # Check session info output.
+  session_info <- app$get_text("#teal-sessionInfo-verbatim_content")
+
+  greplf <- function(text, content = session_info) {
+    grepl(text, content, fixed = TRUE)
+  }
+
+  testthat::expect_true(greplf("R version"))
+  testthat::expect_true(greplf("attached base packages:"))
+  testthat::expect_true(greplf("loaded via a namespace (and not attached):"))
+
+  testthat::expect_true(greplf("shiny"))
+  testthat::expect_true(greplf("teal.slice"))
+  testthat::expect_true(greplf("teal.reporter"))
```

Comment:
Use `expect_match` instead of custom `greplf`

```suggestion
  testthat::expect_match(session_info, "R version", fixed = TRUE)
  testthat::expect_match(session_info, "attached base packages:", fixed = TRUE)
  testthat::expect_match(session_info, "loaded via a namespace (and not attached):", fixed = TRUE)

  testthat::expect_match(session_info, "shiny", fixed = TRUE)
  testthat::expect_match(session_info, "teal.slice", fixed = TRUE)
  testthat::expect_match(session_info, "teal.reporter", fixed = TRUE)
```